swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
+
+from .item import Item
+
+
+
+# WAVE 1
 class Vendor:
-    pass
+    # constructor that is by default empty, but checks for input
+    def __init__(self, inventory = [] ):
+        if inventory :
+            self.inventory = inventory
+        else:
+            self.inventory = []
```

Comment:
We should **NOT** use mutable types like lists as a default parameter. We should avoid doing this at all costs as it will make all Vendor instances share the exact same list. For example:

If we create two Vendors that use the default inventory value of an empty list:
```
asya = Vendor()
audrey = Vendor()
```
This would mean asya and audrey would share the same empty list. Every time asya adds an item to their inventory, the same item would appear in Audrey's (since it's the same list!) We do not want instances to share the same list, especially if they'll be swapping items in the future. 

To prevent this from happening, we can assign inventory to a default value of "None" if no inventory list is provided. Then when we assign the instance variable `self.inventory`, we can use a ternary to determine the value of `inventory`. 

```suggestion
    def __init__(self, inventory=None):
        self.inventory = inventory if inventory is not None else []
```

Reply:
This article uses functions as examples, but the same rule still applies for constructors (since constructors are also functions).

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/